wave_lang/kernel/wave/scheduling/gemm_four_stage_pipelined_scheduling.py

```diff
@@ -0,0 +1,148 @@
+import torch.fx as fx
+from ...ops.wave_ops import get_custom
+from .graph_utils import Edge, sort_graph_by_edge_weight
+from .resources import Operation, get_custom_operation_type
+from enum import Enum
+import math
+
+
+class GemmFourStageStage(Enum):
+    GLOBAL_LOAD = 0
+    LOCAL_STORE = 1
+    LOCAL_LOAD = 2
+    COMPUTE = 3
+    SCHEDULING_NOOP = -1
+
+    # Helper function to get next stage from the current.
+    # If at stage 3 returns itself to prevent crash
+    # since it is final stage.
+    def next(self):
+
+        if self.value == 3:
+            return GemmFourStageStage(3)
+        v = self.value + 1
+        return GemmFourStageStage(v)
+
+
+operation_stage_table = {
+    Operation.READ_SHARED: GemmFourStageStage.LOCAL_LOAD,
+    Operation.WRITE_SHARED: GemmFourStageStage.LOCAL_STORE,
+    Operation.READ_GLOBAL: GemmFourStageStage.GLOBAL_LOAD,
+    Operation.MMA: GemmFourStageStage.COMPUTE,
+    Operation.NOOP: GemmFourStageStage.SCHEDULING_NOOP,
+    Operation.VALU: GemmFourStageStage.COMPUTE,
+    Operation.SHUFFLE: GemmFourStageStage.COMPUTE,
+    Operation.WRITE_GLOBAL: GemmFourStageStage.COMPUTE,
+}
+
+
+def get_scheduling_stage(op: fx.Node) -> GemmFourStageStage:
```

Comment:
Should the return type be Operation? Also seems like you could reuse this function from prefetch_scheduling.py?